app/models/operation.rb

```diff
@@ -1,3 +1,25 @@
 class Operation < ApplicationRecord
   belongs_to :tank
+  belongs_to :family, required: false
+
+  validates :family, presence: true, if: :add_family?
+
+  def perform
+    case action.to_sym
```

Comment:
We call `action&.to_sym` a few different places, we may want to extract the following method:

```
def action
  action&.to_sym
end
```